iati/core/tests/test_rulesets.py

```diff
@@ -142,29 +142,26 @@ def test_rule_class_cannot_be_instantiated_directly_with_name(self):
 class TestRuleSubclasses(object):
     """A container for tests relating to all Rule subclasses."""
 
-    rule_constructors = list(map(iati.core.rulesets.locate_constructor_for_rule_type, iati.core.rulesets._VALID_RULE_TYPES))
-    """A list of constructors for the various types of Rule."""
+    @pytest.fixture(params=[
+        iati.core.rulesets.locate_constructor_for_rule_type(rule_type) for rule_type in iati.core.rulesets._VALID_RULE_TYPES
+    ])
+    def rule_constructor(self, request):
+        """List constructors for the various types of Rule."""
+        return request.param
 
-    @pytest.mark.parametrize("rule_constructor", rule_constructors)
     def test_rule_init_no_parameters(self, rule_constructor):
         """Check that a Rule cannot be created when no parameters are given."""
         with pytest.raises(TypeError):
             rule_constructor()
 
-    @pytest.mark.parametrize("rule_constructor", rule_constructors)
     @pytest.mark.parametrize("xpath_base", iati.core.tests.utilities.find_parameter_by_type(['str'], False))
     def test_rule_init_invalid_xpath_base(self, rule_constructor, xpath_base):
-        """Check that a Rule cannot be created when xpath_base is not a string.
-
-        Todo:
-            Ensure the case is valid - an empty dictionary isn't.
-        """
+        """Check that a Rule cannot be created when xpath_base is not a string."""
         case = dict()
```

Comment:
`case` is invalid. As such, it cannot be asserted that it is the invalid `xpath_base` that is causing the `ValueError` to be raised.

One should be valid, the other invalid.